conversion/conversion.go

```diff
@@ -826,6 +834,9 @@ func generateName(prefix string) (string, error) {
 // GetClient returns new spanner client.
 func GetClient(db string) (*sp.Client, error) {
 	ctx := context.Background()
+	if endpoint := os.Getenv("SPANNER_API_ENDPOINT"); endpoint != "" {
+		return sp.NewClient(ctx, db, option.WithEndpoint(endpoint))
+	}
 	return sp.NewClient(ctx, db)
 }
```

Comment:
Consider moving NewDatabaseAdminClientWithEndpoint (or NewDatabaseAdminClient, please see comment above) here and renaming GetClient to NewDatabaseClientWithEndPoint (or NewDatabaseClient correspondingly) for consistency.

Reply:
Kept GetClient but turned it into a wrapper that just fetches the relevant ctx and passes it to a new function NewSpannerClient.  This means that New*Client can have consistent signatures with their downstream counterparts.